### PR TITLE
fix: smart contracts migrations

### PR DESCRIPTION
**Pull request by antmat**
This PR is intended to provide more clear and stable migration process:
  - migration artifacts now are saved in migration_artifacts folder to maintain migration data
  - migration contract from truffle suit is used.
  - all current migrations are packed into one, as we start from the beginning
  - usefull makefile targets introduced 
  - npm ci command is used for build stability

### CODE REVIEW

File: blockchain/source/contracts/Migrations.sol [hunk 1]
```diff
@@ -12,11 +12,11 @@ contract Migrations {
         owner = msg.sender;
     }
 
-    function setCompleted(uint completed) public restricted {
+    function setCompleted(uint completed) restricted {
```
Comment (quasisamurai) [inline blockchain/source/contracts/Migrations.sol]: why removed visibility?

File: blockchain/source/scripts/test.sh [hunk 2]
```diff
@@ -16,7 +16,7 @@ cleanup() {
 if [ "$SOLIDITY_COVERAGE" = true ]; then
   testrpc_port=8555
 else
-  testrpc_port=8535
+  testrpc_port=8525
```
Comment (quasisamurai) [inline blockchain/source/scripts/test.sh]: change port back, plz

**Comment by sokel** [comment]
image

File: blockchain/source/migrations/2_deploy_v1.js [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        return 5;
+    }
+}
+
+function MSOwners (network, accounts) {
+    if (network === 'dev_main' || network === 'dev_side') {
+        return accounts;
+    } else {
+        accounts = [
+            '0xdaec8F2cDf27aD3DF5438E5244aE206c5FcF7fCd',
```
Comment (sokel) [inline blockchain/source/migrations/2_deploy_v1.js]: inactive owner address

File: blockchain/source/migrations/2_deploy_v1.js [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+            '0xd9a43e16e78c86cf7b525c305f8e72723e0fab5e',
+            '0x72cb2a9AD34aa126fC02b7d32413725A1B478888',
+            '0x1f50Be5cbFBFBF3aBD889e17cb77D31dA2Bd7227',
+            '0xe062C67207F7E478a93EF9BEA39535d8EfFAE3cE',
+            '0x5fa359a9137cc5ac2a85d701ce2991cab5dcd538',
+            '0x7aa5237e0f999a9853a9cc8c56093220142ce48e',
+            '0xd43f262536e916a4a807d27080092f190e25d774',
+            '0xdd8422eed7fe5f85ea8058d273d3f5c17ef41d1c',
+        ];
+        // antmat's key
```
Comment (sokel) [inline blockchain/source/migrations/2_deploy_v1.js]: pls share this key for others

File: blockchain/source/contracts/Migrations.sol (hunk 1 above)
Comment (antmat) [reply blockchain/source/contracts/Migrations.sol]: truffle doesn't properly detect API of the migrations contract in this case

File: blockchain/source/migrations/2_deploy_v1.js (hunk 3 above)
Comment (antmat) [reply blockchain/source/migrations/2_deploy_v1.js]: it was there previously, so I didn't change it

File: blockchain/source/Makefile [hunk 5]
```diff
@@ -33,34 +33,55 @@ lint_js:
...
+endif
+ifeq (${PRV_KEY_SIDE},)
+	@exit 1
+endif
+ifeq (${MS_PRV_KEY},)
+	@exit 1
+endif
+
+migrate_dev:
+	PRV_KEY_MAIN=4f3edf983ac636a65a842ce7c78d9aa706d3b113bce9c46f30d7d21715b23b1d \
```
Comment (3Hren) [inline blockchain/source/Makefile]: Decompose into a variable?

File: blockchain/source/Makefile [hunk 6]
```diff
@@ -33,34 +33,55 @@ lint_js:
...
+	PRV_KEY_SIDE=4f3edf983ac636a65a842ce7c78d9aa706d3b113bce9c46f30d7d21715b23b1d \
+	MS_PRV_KEY=4f3edf983ac636a65a842ce7c78d9aa706d3b113bce9c46f30d7d21715b23b1d \
+	${TRUFFLE} migrate --network=dev_side
+	PRV_KEY_MAIN=4f3edf983ac636a65a842ce7c78d9aa706d3b113bce9c46f30d7d21715b23b1d \
```
Comment (3Hren) [inline blockchain/source/Makefile]: Repeated part.

File: blockchain/source/contracts/Migrations.sol (hunk 1 above)
Comment (sokel) [reply blockchain/source/contracts/Migrations.sol]: @quasisamurai its ok, bro

File: blockchain/source/scripts/dev.sh [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+pids=()
+
+blockTime=$1
+if [[ -n "$blockTime" ]]; then
+    blockTime="-b $blockTime"
+fi
+
+cleanup() {
+  echo "huy"
```
Comment (sokel) [inline blockchain/source/scripts/dev.sh]: =)

**Pull request merged**